From 757a5fe9fb5ce53d62bfafe8f4c47dc2ece95b34 Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Fri, 18 Dec 2020 17:35:19 +0800
Subject: [PATCH 26/75] Revert "matroska-mux: Fix incorrect rounding of
 timestamps"

this commit will cause our mkv parser dead loop while seeking keyframe

This reverts commit ce802f033c8cee4f923c0d893dda0efd0d003689.

Upstream-Status: Pending

(cherry picked from commit fdf888b4af97dd779d77ffa3f18e740842989532)
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 gst/matroska/matroska-mux.c | 32 +++++---------------------------
 1 file changed, 5 insertions(+), 27 deletions(-)

diff --git a/gst/matroska/matroska-mux.c b/gst/matroska/matroska-mux.c
index 0fccd55..43f3584 100644
--- a/gst/matroska/matroska-mux.c
+++ b/gst/matroska/matroska-mux.c
@@ -4017,7 +4017,6 @@ gst_matroska_mux_write_data (GstMatroskaMux * mux, GstMatroskaPad * collect_pad,
   GstBuffer *hdr;
   guint64 blockgroup;
   gboolean write_duration;
-  guint64 cluster_time_scaled;
   gint16 relative_timestamp;
   gint64 relative_timestamp64;
   guint64 block_duration, duration_diff = 0;
@@ -4135,8 +4134,6 @@ gst_matroska_mux_write_data (GstMatroskaMux * mux, GstMatroskaPad * collect_pad,
         gst_pad_push_event (mux->srcpad, mux->force_key_unit_event);
         mux->force_key_unit_event = NULL;
       }
-      cluster_time_scaled =
-          gst_util_uint64_scale (buffer_timestamp, 1, mux->time_scale);
 
       mux->prev_cluster_size = ebml->pos - mux->cluster_pos;
       mux->cluster_pos = ebml->pos;
@@ -4144,44 +4141,25 @@ gst_matroska_mux_write_data (GstMatroskaMux * mux, GstMatroskaPad * collect_pad,
       mux->cluster =
           gst_ebml_write_master_start (ebml, GST_MATROSKA_ID_CLUSTER);
       gst_ebml_write_uint (ebml, GST_MATROSKA_ID_CLUSTERTIMECODE,
-          cluster_time_scaled);
+          gst_util_uint64_scale (buffer_timestamp, 1, mux->time_scale));
       GST_LOG_OBJECT (mux, "cluster timestamp %" G_GUINT64_FORMAT,
           gst_util_uint64_scale (buffer_timestamp, 1, mux->time_scale));
       gst_ebml_write_flush_cache (ebml, is_video_keyframe
           || is_audio_only, buffer_timestamp);
+      mux->cluster_time = buffer_timestamp;
       gst_ebml_write_uint (ebml, GST_MATROSKA_ID_PREVSIZE,
           mux->prev_cluster_size);
-      /* cluster_time needs to be identical in value to what's stored in the
-       * matroska so we need to have it with the same precision as what's
-       * possible with the set timecodescale rather than just using the
-       * buffer_timestamp.
-       * If this is not done the rounding of relative_timestamp will be
-       * incorrect and possibly making the timestamps get out of order if tw
-       * buffers arrive at the same millisecond (assuming default timecodescale
-       * of 1ms) */
-      mux->cluster_time =
-          gst_util_uint64_scale (cluster_time_scaled, mux->time_scale, 1);
     }
   } else {
     /* first cluster */
-    cluster_time_scaled =
-        gst_util_uint64_scale (buffer_timestamp, 1, mux->time_scale);
+
     mux->cluster_pos = ebml->pos;
     gst_ebml_write_set_cache (ebml, 0x20);
     mux->cluster = gst_ebml_write_master_start (ebml, GST_MATROSKA_ID_CLUSTER);
     gst_ebml_write_uint (ebml, GST_MATROSKA_ID_CLUSTERTIMECODE,
-        cluster_time_scaled);
+        gst_util_uint64_scale (buffer_timestamp, 1, mux->time_scale));
     gst_ebml_write_flush_cache (ebml, TRUE, buffer_timestamp);
-    /* cluster_time needs to be identical in value to what's stored in the
-     * matroska so we need to have it with the same precision as what's
-     * possible with the set timecodescale rather than just using the
-     * buffer_timestamp.
-     * If this is not done the rounding of relative_timestamp will be
-     * incorrect and possibly making the timestamps get out of order if tw
-     * buffers arrive at the same millisecond (assuming default timecodescale
-     * of 1ms) */
-    mux->cluster_time =
-        gst_util_uint64_scale (cluster_time_scaled, mux->time_scale, 1);
+    mux->cluster_time = buffer_timestamp;
   }
 
   /* We currently write index entries for all video tracks or for the audio
-- 
2.43.0

